### DISCUSSION TITLE
Why CrossfadePlugin stop looping gif in GlideImage.

### DISCUSSION THREAD

**Discussion by AndreyMedal** [skydoves/landscapist#327] [Q&A]
I added `+CrossfadePlugin()` for my simple `GlideImage` with gif and it stop looping after `CrossFade` animation is finish. 
Any Ideas how to use `CrossfadePlugin` with gifs?

**Comment by faraz152** [comment]
This is a fundamental incompatibility between `CrossfadePlugin` and animated GIFs in Glide.

**Root cause:** `CrossfadePlugin` wraps Glide's loaded drawable in a `DrawableCrossFadeTransition`. Once the crossfade animation completes, Glide replaces the transitioning drawable with the final frame drawable — and at that point the `GifDrawable`'s animation loop is **not restarted**. The GIF plays through the crossfade, then freezes on whatever frame was shown when the transition ended.

**Solution — don't use `CrossfadePlugin` for GIFs. Use a Compose-side fade instead:**

```kotlin
var isLoaded by remember { mutableStateOf(false) }

AnimatedVisibility(
    visible = isLoaded,
    enter = fadeIn(animationSpec = tween(durationMillis = 300)),
) {
    GlideImage(
        imageModel = { gifUrl },
        modifier = Modifier.fillMaxSize(),
    )
}

// Trigger fade-in once Glide finishes loading
GlideImage(
    imageModel = { gifUrl },
    modifier = Modifier.fillMaxSize(),
    onImageStateChanged = { state ->
        if (state is GlideImageState.Success) isLoaded = true
    },
)
```

Or, if you only need the crossfade on the **first load** (placeholder → image) and the GIF is already cached on subsequent views, set the transition on the request builder instead of using CrossfadePlugin:

```kotlin
GlideImage(
    imageModel = { gifUrl },
    imageOptions = ImageOptions(
        requestBuilder = {
            // Crossfade only applies during the loading transition;
            // after it completes, Glide hands back the real GifDrawable
            // which continues looping normally.
            transition(DrawableTransitionOptions.withCrossFade(300))
        }
    )
)
```

Note: `DrawableTransitionOptions.withCrossFade()` set directly on the `RequestBuilder` behaves differently from `CrossfadePlugin` — it may still pause the GIF depending on Glide version. If GIF looping is critical, the `AnimatedVisibility` approach is the most reliable.
